src/site/content/en/lighthouse-best-practices/no-document-write/index.md

```diff
@@ -1,47 +1,61 @@
 ---
 layout: post
-title: Page uses document.write()
+title: Uses document.write()
 description: |
-  Learn about `no-document-write` audit.
+  Learn how to speed up your page's load time by avoiding  document.write().
 web_lighthouse:
   - no-document-write
+updated: 2019-08-28
 ---
 
 For users on slow connections,
 such as 2G, 3G, or slow Wi-Fi,
-external scripts dynamically injected via `document.write()`
+third-party scripts dynamically injected via `document.write()`
 can delay the display of main page content by tens of seconds.
-Lighthouse lists out every call to `document.write()`:
+Chrome therefore blocks the execution of these scripts.
+
+## How the Lighthouse `document.write()` audit fails
+
+[Lighthouse](https://developers.google.com/web/tools/lighthouse/) flags
+all calls to `document.write()`:
 
 <figure class="w-figure">
-  <img class="w-screenshot w-screenshot--filled" src="no-document-write.png" alt="Lighthouse audit showing usage of document.write">
-  <figcaption class="w-figcaption">
-    Page uses <code>document.write()</code>.
-  </figcaption>
+  <img class="w-screenshot" src="no-document-write.png" alt="Lighthouse audit showing usage of document.write">
 </figure>
 
-See [Intervening against `document.write()`](https://developers.google.com/web/updates/2016/08/removing-document-write)
-to learn more.
+Lighthouse reports calls to `document.write()` because it's often used to inject
```

Comment:
tbh this topic is confusing.. 

1. document.write() has terrible performance characteristics for everyone. there are some cases where it's extremely bad, but there's no real good reason to use the feature. (it sucks even if you're loading a page on a superfast connection, etc)
1. the intervention (when it's applied) means that some particular* calls to document.write() will just not do anything. they're just no-ops. if this happens, there will be a message in the devtools console saying so. lighthouse doesn't report here on that situation, though it would be in the errors-in-console audit
   - sometimes the intervention doesn't block the call (because the connection wasn't slow) but it will emit a console warning. Lighthouse doesn't highlight this in either audit. (it probably should)
1.  this lighthouse audit captures calls to document.write which the intervention didn't stop. so the calls did indeed succeed. however at the same time they emit a console message (which is hidden by default to devtools users) saying "Parser was blocked due to document.write". And those are the cases this particular audit reports on.